__tests__/unit/reference.spec.js

```diff
@@ -131,4 +222,67 @@ describe('Reference testing suite', () => {
   // .transaction()
 
   // .update()
+  it('should throw an error when update is called with no arguments', done => {
+    ref = new Reference(app);
+
+    try {
+      ref.update();
+    } catch (error) {
+      expect(error.message).toBe('Undefined is not allowed in the data tree.');
+      done();
+    }
+  });
+
+  it('should delete value when update is called with null', async () => {
+    ref = new Reference(app);
+
+    await ref.update(null);
+    let snapshot = await ref.once('value');
+
+    expect(snapshot.val()).toBe(null);
+  });
+
+  it('should update the value of the ref when update is called with a new non-object value', async done => {
+    ref = new Reference(app)
+
+    let dataSet = {
+      'foo': 'bar',
+      'bar': 'foo'
+    }
+
+    await ref.set(dataSet);
+    let snapshot = await ref.once('value');
+    expect(snapshot.val()).toEqual(dataSet);
+
+    ref.child('foo').update('foo', async () => {
+
+      let fooSnapshot = await ref.child('foo').once('value');
+
+      expect(fooSnapshot.val()).toBe('foo');
+      done();
+    });
+  });
+
+  it('should update the value of the ref when update is alled with an object value', async done => {
```

Comment:
This test seems to be failing as a result of an issue with code execution not completing past [this](https://github.com/sebasgarcep/firebase-admin-mock/pull/6/files#diff-9306dd252215d2a33affe15ec19e6dc7R383) point. I'm not well versed enough with `reduce` functions to know exactly what is going on.

cc: @sebasgarcep